### DISCUSSION TITLE
Trying to configure cmp-cmdline made both native and nvim-cmp menus appear

### DISCUSSION THREAD

**Discussion by ericukk** [LazyVim/LazyVim#2373] [Q&A] [answered]
![image](https://github.com/LazyVim/LazyVim/assets/156708809/c76ab326-c431-4a69-a814-f9f73225170c)
```lua
  {
    "hrsh7th/cmp-cmdline",
    dependencies = { "hrsh7th/nvim-cmp" },
    config = function(_, _)
      local cmp = require("cmp")
      cmp.setup.cmdline(":", {
        mapping = cmp.mapping.preset.insert(),
        sources = cmp.config.sources({
          { name = "path" },
        }, {
          { name = "cmdline" },
        }),
      })
    end,
  },

```
Could someone explain to me why this doesn't work?
I'm trying to follow the [official doc](https://github.com/hrsh7th/nvim-cmp#recommended-configuration) and translate it to lazy.nvim. The only dependency is to run `cmp.setup.cmdline` after nvim-cmp's setup, which should already be enforced by `dependencies = { "hrsh7th/nvim-cmp" }`, right?
 
One way that worked for me is to put `cmp.setup.cmdline` inside "hrsh7th/nvim-cmp" config function. But I wouldn't want to override the default `config` function that LazyVim already provides.

**Comment by jyuan0** [comment]
Maybe they're both appearing because the docs have `mapping = cmp.mapping.preset.cmdline()`, whereas you have `mapping = cmp.mapping.preset.insert()`?

> The only dependency is to run cmp.setup.cmdline after nvim-cmp's setup, which should already be enforced by dependencies = { "hrsh7th/nvim-cmp" }, right?

Yes. The default LazyVim `config` function is also pretty small.

**Reply by dpetka2001** [reply]
Edit: oops sorry, was meant to answer to the OP.

**Reply by jyuan0** [reply]
No worries. Thanks for all the help you give @dpetka2001!

**Comment by dpetka2001** [comment, marked as answer]
You can check [here](https://github.com/LazyVim/LazyVim/discussions/2072) for a solution. Test it if it works for you.

**Reply by ericukk** [reply]
But how can I write a config fn without overriding the LazyVim one?

**Reply by dpetka2001** [reply]
You're not overwriting the Lazyvim default. Lazyvim has config function for `nvim-cmp`. This one is for `cmp-cmdline`. They're not the same.

**Reply by ericukk** [reply]
is there any way to it though? I'd like to add some autocmds to nvim-lspconfig, but it already has a huge config.

**Reply by dpetka2001** [reply]
You can't merge `config` functions. There can be only one. Give an example of what you'd like to do to see if there's probably a better way to do it or not.
